cookbooks/cdo-ruby/recipes/ruby-build.rb

```diff
@@ -6,14 +6,14 @@
 
 # Arbitrarily use the latest version of ruby build at time this code was
 # written; this probably doesn't matter, but we need to pick something.
-RUBY_BUILD_VERSION = '20221225'.freeze
+RUBY_BUILD_VERSION = '20240727'.freeze
 
-remote_file '/tmp/ruby-build.tar.gz' do
+remote_file "/tmp/ruby-build-#{RUBY_BUILD_VERSION}.tar.gz" do
   source "https://github.com/rbenv/ruby-build/archive/refs/tags/v#{RUBY_BUILD_VERSION}.tar.gz"
   action :create_if_missing
 end
 
-archive_file '/tmp/ruby-build.tar.gz' do
+archive_file "/tmp/ruby-build-#{RUBY_BUILD_VERSION}.tar.gz" do
```

Comment:
The archive needs the version in the filename or Chef won't try the new version (if the file already exists from the old version).